### PR TITLE
Fix memcached error parsing

### PR DESCRIPTION
**Pull request by tonyd3**
Problem
parseErrorMessage only returns the last string, instead of the entire message.
i.e. FAT [20150721-20:26:43.792] memcache: com.twitter.finagle.memcached.protocol.ServerError: object

Solution
Parse all of the tokens instead of the head.

Result
com.twitter.finagle.memcached.protocol.ServerError: out of memory storing object

### CODE REVIEW

File: finagle-memcachedx/src/main/scala/com/twitter/finagle/memcachedx/protocol/text/client/DecodingToResponse.scala [hunk 1]
```diff
@@ -73,6 +73,6 @@ class DecodingToResponse extends AbstractDecodingToResponse[Response] {
   }
 
   private[this] def parseErrorMessage(tokens: Seq[Buf]) =
-    tokens.lastOption map { case Buf.Utf8(s) => s } getOrElse("")
+    if (tokens.isEmpty) "" else tokens.tail map { case Buf.Utf8(s) => s  } mkString(" ")
```
Comment (vkostyukov) [inline finagle-memcachedx/src/main/scala/com/twitter/finagle/memcachedx/protocol/text/client/DecodingToResponse.scala]: I believe the style is `tokens.tail.map {}.mkString(" ") `.

Comment (vkostyukov) [inline finagle-memcachedx/src/main/scala/com/twitter/finagle/memcachedx/protocol/text/client/DecodingToResponse.scala]: And why the empty `tokens` considered as a special case? `Nil.mkString(" ")` will produce a "" string as expected.

Comment (tonyd3) [inline finagle-memcachedx/src/main/scala/com/twitter/finagle/memcachedx/protocol/text/client/DecodingToResponse.scala]: Doesn't tokens.tail throw an exception if it's empty?

Comment (tonyd3) [inline finagle-memcachedx/src/main/scala/com/twitter/finagle/memcachedx/protocol/text/client/DecodingToResponse.scala]: I think calling tokens.tail on an empty seq throws an UnsupportedOperationException("empty.tail")

Comment (vkostyukov) [inline finagle-memcachedx/src/main/scala/com/twitter/finagle/memcachedx/protocol/text/client/DecodingToResponse.scala]: Right. Just use `drop(1)` then.

Comment (tonyd3) [inline finagle-memcachedx/src/main/scala/com/twitter/finagle/memcachedx/protocol/text/client/DecodingToResponse.scala]: Done

**Comment by vkostyukov** [comment]
:+1: Thanks @tonyd3!

**Comment by mosesn** [comment]
Can you add a test?  LGTM though.

**Comment by dschobel** [comment]
LGTM and +1 for adding test coverage.

**Comment by vkostyukov** [comment]
@tonyd3 any progress with the test? Let us know if we can help here.

**Comment by tonyd3** [comment]
@vkostyukov just a busy day dealing with some other stuff, will get to it tonight or tomorrow.

**Comment by tonyd3** [comment]
added tests

**Comment by vkostyukov** [comment]
Thanks @tonyd3! Looks great!

File: finagle-memcachedx/src/test/scala/com/twitter/finagle/memcachedx/unit/protocol/text/client/DecodingToResponseTest.scala [hunk 2]
```diff
@@ -89,4 +89,15 @@ class DecodingToResponseTest extends FunSuite {
     assert(error.cause.getMessage() === errorMessage)
   }
 
+  test("parseResponse SERVER_ERROR") {
+    val context = new Context
+    import context._
+
+    val errorMessage = "sad panda error"
+    val buffer = Tokens( Seq("SERVER_ERROR", "sad", "panda", "error") map { Buf.Utf8(_) } )
```
Comment (dschobel) [inline finagle-memcachedx/src/test/scala/com/twitter/finagle/memcachedx/unit/protocol/text/client/DecodingToResponseTest.scala]: no postfix ops please

**Comment by dschobel** [comment]
lgtm after style fix

File: finagle-memcachedx/src/test/scala/com/twitter/finagle/memcachedx/unit/protocol/text/client/DecodingToResponseTest.scala [hunk 3]
```diff
@@ -89,4 +89,15 @@ class DecodingToResponseTest extends FunSuite {
...
+    val error = decodingToResponse.decode(null, null, buffer).asInstanceOf[protocol.Error]
+    assert(error.cause.getClass === classOf[ServerError])
```
Comment (mosesn) [inline finagle-memcachedx/src/test/scala/com/twitter/finagle/memcachedx/unit/protocol/text/client/DecodingToResponseTest.scala]: no need for triple equals, == is fine.  scalatest 2.x has an assert macro that's the :bomb:

**Comment by mosesn** [comment]
LGTM

**Comment by tonyd3** [comment]
updated

**Comment by vkostyukov** [comment]
Nice! I'm pulling this for the internal review.

**Comment by vkostyukov** [comment]
This has been merged internally. Thanks @tonyd3!

**Pull request closed without merging**